From cc51b3b15b0c4ca82b5c22a30d7b69456ee1c4a7 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:45 -0400
Subject: [scsi] hpsa: do not ignore failure of sense controller parameters command

Message-id: <1410186734-19986-55-git-send-email-thenzl@redhat.com>
Patchwork-id: 92065
O-Subject: [RHEL6.6 e-stor PATCH 54/83] hpsa: do not ignore failure of sense controller parameters command
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Signed-off-by: Joe Handzik <joseph.t.handzik@hp.com>
Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 96444fbbbf3b55a9cd1e337cc5c8c7d04bb364b7)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 7e924d0..3d147e1 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -2989,12 +2989,12 @@ static int hpsa_hba_mode_enabled(struct ctlr_info *h)
   GFP_KERNEL);
 
  if (!ctlr_params)
-  return 0;
+  return -ENOMEM;
  rc = hpsa_bmic_ctrl_mode_sense(h, RAID_CTLR_LUNID, 0, ctlr_params,
   sizeof(struct bmic_controller_parameters));
- if (rc != 0) {
+ if (rc) {
   kfree(ctlr_params);
-  return 0;
+  return rc;
  }
 
  hba_mode_enabled =
@@ -3041,6 +3041,8 @@ static void hpsa_update_scsi_devices(struct ctlr_info *h, int hostno)
  memset(lunzerobits, 0, sizeof(lunzerobits));
 
  rescan_hba_mode = hpsa_hba_mode_enabled(h);
+ if (rescan_hba_mode < 0)
+  goto out;
 
  if (!h->hba_mode_enabled && rescan_hba_mode)
   dev_warn(&h->pdev->dev, "HBA mode enabled\n");
-- 
1.7.1